cpp/include/raft/spatial/knn/detail/ivf_pq_search.cuh

```diff
@@ -393,11 +393,27 @@ void postprocess_neighbors(IdxT* neighbors,  // [n_queries, topk]
                                            topk);
 }
 
+template <typename T>
+HDI auto undo_normalization(const float& x, bool squared)
+  -> std::enable_if_t<std::is_same_v<T, float>, float>
+{
+  return x;
+};
+
+template <typename T>
+HDI auto undo_normalization(const float& x, bool squared)
```

Comment:
I think it would be less code if we didn't have this function at all? :) If at some point we add more metric types, we will have to continue adding more parameter here (power parameter, etc.). Also just logically `undo_normalization` would mean different things here for different metrics.